python/dgl/graphbolt/subgraph_sampler.py

```diff
@@ -1,22 +1,147 @@
 """Subgraph samplers"""
 
+from collections import defaultdict
+from typing import Dict
+
 from torchdata.datapipes.iter import Mapper
 
+from .impl import SampledSubgraphImpl
+
+from .link_prediction_block import LinkPredictionBlock
+from .node_classification_block import NodeClassificationBlock
+from .utils import unique_and_compact, unique_and_compact_node_pairs
+
 
 class SubgraphSampler(Mapper):
-    """A subgraph sampler.
+    """A subgraph sampler used to sample a subgraph from a given set of nodes
+    from a larger graph."""
+
+    def __init__(
+        self,
+        datapipe,
+        fanouts,
+        replace=False,
+        prob_name=None,
```

Comment:
This should be specific to NeighborSampler rather than SubgraphSampler

Reply:
Got you, that make sense